src/Microsoft.SymbolStore/KeyGenerators/PEFileKeyGenerator.cs

```diff
@@ -120,79 +126,104 @@ public override IEnumerable<SymbolStoreKey> GetKeys(KeyTypeFlags flags)
+                string[] clrSOSFiles = new string[] { "SOS.dll" };
+                IEnumerable<string> longNameSOSFiles = GetFilesLongNameVariants(SosFileName);
+                return clrSOSFiles.Concat(longNameSOSFiles);
+            }
 
-                List<string> hostArchitectures = new();
-                string targetArchitecture = null;
+            return default;
+        }
 
-                ImageFileMachine machine = (ImageFileMachine)_peFile.FileHeader.Machine;
-                switch (machine)
-                {
-                    case ImageFileMachine.Amd64:
-                        targetArchitecture = "amd64";
-                        break;
-
-                    case ImageFileMachine.I386:
-                        targetArchitecture = "x86";
-                        break;
-
-                    case ImageFileMachine.ArmNT:
-                        targetArchitecture = "arm";
-                        hostArchitectures.Add("x86");
-                        break;
-
-                    case ImageFileMachine.Arm64:
-                        targetArchitecture = "arm64";
-                        hostArchitectures.Add("amd64");
-                        break;
-                }
+        private IEnumerable<string> GetDACFiles(string runtimeFileName)
+        {
+            if (runtimeFileName == CoreClrFileName)
+            {
+                string[] coreClrDACFiles = new string[] { "mscordaccore.dll", "mscordbi.dll" };
+                IEnumerable<string> longNameDACFiles = GetFilesLongNameVariants(CoreClrDACFileName);
+                return coreClrDACFiles.Concat(longNameDACFiles);
+            }
 
-                if (targetArchitecture != null)
-                {
-                    hostArchitectures.Add(targetArchitecture);
+            if (runtimeFileName == ClrFileName)
+            {
+                string[] clrDACFiles = new string[] { "mscordacwks.dll", "mscordbi.dll" };
+                IEnumerable<string> longNameDACFiles = GetFilesLongNameVariants(ClrDACFileName);
+                return clrDACFiles.Concat(longNameDACFiles);
+            }
 
-                    foreach (string hostArchitecture in hostArchitectures)
-                    {
-                        string buildFlavor = "";
+            return default;
+        }
 
-                        if ((fileVersion.FileFlags & FileInfoFlags.Debug) != 0)
-                        {
-                            if ((fileVersion.FileFlags & FileInfoFlags.SpecialBuild) != 0)
-                            {
-                                buildFlavor = ".dbg";
-                            }
-                            else
-                            {
-                                buildFlavor = ".chk";
-                            }
-                        }
+        private IEnumerable<string> GetFilesLongNameVariants(string fileWithLongNameVariant) => GetFilesLongNameVariants(new string[] { fileWithLongNameVariant });
```

Comment:
This overload isn't needed.  Just add `params` to the other overload.  e.g. `private IEnumerable<string> GetFilesLongNameVariants(params string[] filesWithLongNameVariant)`.

Reply:
Thanks! Fixed